src/main/java/ee/tuleva/onboarding/payment/provider/montonio/MontonioOrderCreator.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Locale;
+import java.util.Objects;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class MontonioOrderCreator {
+
+  private final Clock clock;
+
+  private final PaymentInternalReferenceService paymentInternalReferenceService;
+
+  private final MontonioPaymentChannelConfiguration montonioPaymentChannelConfiguration;
+
+  private final LocaleService localeService;
+
+  @Value("${payment-provider.use-fake-notification-url}")
+  private Boolean useFakeNotificationsUrl;
+
+  @Value("${api.url}")
+  private String apiUrl;
+
+  @Value("${payment.member-fee}")
+  private BigDecimal memberFee;
+
+  @Value("${payment.member-fee-test-personal-code}")
+  private String memberFeeTestPersonalCode;
+
+  private static String getEpisPaymentDescription(PaymentData paymentData) {
+    if (paymentData.getType() == PaymentType.MEMBER_FEE) {
+      return String.format("member:%s", paymentData.getRecipientPersonalCode());
+    }
+
+    return String.format(
+        "30101119828, IK:%s, EE3600001707", paymentData.getRecipientPersonalCode());
+  }
+
+  @SneakyThrows
+  public MontonioOrder getOrder(PaymentData paymentData, Person person) {
+    MontonioPaymentChannel paymentChannelConfiguration =
+        montonioPaymentChannelConfiguration.getPaymentProviderChannel(
+            paymentData.getPaymentChannel());
+
+    BigDecimal amount = getPaymentAmount(paymentData);
+    Currency currency = paymentData.getCurrency();
+
+    return MontonioOrder.builder()
+        .accessKey(paymentChannelConfiguration.getAccessKey())
+        .merchantReference(paymentInternalReferenceService.getPaymentReference(person, paymentData))
+        .returnUrl(getPaymentSuccessReturnUrl(paymentData.getType()))
+        .notificationUrl(getNotificationUrl())
+        .grandTotal(amount)
+        .currency(currency)
+        .exp(clock.instant().getEpochSecond() + 600)
+        .locale(getLanguage())
+        .payment(
+            MontonioPaymentMethod.builder()
+                .amount(amount)
+                .currency(currency)
+                .methodOptions(
+                    MontonioPaymentMethodOptions.builder()
+                        .preferredProvider(paymentChannelConfiguration.getBic())
+                        .preferredLocale(getLanguage())
+                        .paymentDescription(getEpisPaymentDescription(paymentData))
+                        .build())
+                .build())
+        .build();
+  }
+
+  private String getPaymentSuccessReturnUrl(PaymentType paymentType) {
+    if (paymentType == PaymentType.MEMBER_FEE) {
+      return apiUrl + "/payments/member-success";
+    }
+
+    return apiUrl + "/payments/success";
+  }
+
+  private BigDecimal getPaymentAmount(PaymentData paymentData) {
+    if (paymentData.getType() == PaymentData.PaymentType.MEMBER_FEE) {
```

Comment:
You can static import `PaymentData.PaymentType`for better readability:
`paymentData.getType() == MEMBER_FEE`